### DISCUSSION TITLE
How to set EXTENSION_GALLERY env variable when installing via brew

### DISCUSSION THREAD

**Discussion by vladfrangu** [coder/code-server#5604] [Q&A] [answered]
Hey!

I'm trying to setup code-server on my laptop but I am unable to figure out how to set the EXTENSION_GALLERY env variable to make code-server use a different extension server (as open-vsx is currently down). From what I gathered, either setting it in a `product.json` or in the env variable should be enough (keyword being _should_). I tried adding it to my path before calling `brew services restart code-server`, I tried running `brew services start code-server` with the env variable, tried adding a product.json file in `~/.config/code-server` and in `~/.local/share/code-server` *and* `~/.local/share/code-server/Machine` but that also did not work.

Any ideas what else I can do to set this up?
Thanks!

**Comment by jsjoeio** [comment, marked as answer]
I did some looking around and couldn't find anything about passing environment variables to `brew services`. I only found this: https://github.com/Homebrew/homebrew-services/issues/104

I think you'd have to manually edit the file that `brew` uses when they start a session. 

<img width="774" alt="image" src="https://user-images.githubusercontent.com/3806031/193331481-d99e0418-5479-4e37-bb0c-0d65c3912585.png">


Or you might be able to add it to `PATH` and code-server should read it from there. Let me know if that works!

**Reply by vladfrangu** [reply]
Sweet! That totally worked! Only downside is that the server didn't log that it got a custom extension gallery configured 😓

Thanks a bunch!

**Reply by jsjoeio** [reply]
Darn it 😢 Not 100% sure if that's a bug or not (cc @code-asher) but happy it worked!

**Reply by code-asher** [reply]
I think we no longer log that there is a custom gallery set but we could add one at the debug level or something.  (Or maybe info level would be better.)
